### PR TITLE
Support aws opentelemetry distro in Lambda Python Runtimes

### PR DESCRIPTION
**Pull request by wangzlei**
*Issue #, if available:*

*Description of changes:*

Support aws opentelemetry distro in AWS Lambda Python Runtimes. The PR contains:

* Build aws opentelemetry distro Lambda python layer
Run `build-lambda-layer.sh` in `aws-otel-python-instrumentation/lambda-layer/src`

* Unit test for aws opentelemetry distro Lambda python layer
Run `tox` in `aws-otel-python-instrumentation/lambda-layer/src`

* Deploy the aws opentelemetry distro Lambda python layer with an Application Signals enabled Sample App in personal AWS account.
Run `build.sh` in `aws-otel-python-instrumentation/lambda-layer`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lambda-layer/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+pushd src || exit
+rm -rf build
+./build-lambda-layer.sh
+popd || exit
+
+pushd sample-apps || exit
+rm -rf build
+./package-lambda-function.sh
+popd || exit
+
```
Comment (mxiamxia) [inline lambda-layer/build.sh:13]: Q: why does the `sample-apps` need to be built while building the layer artifact?

Comment (mxiamxia) [reply lambda-layer/build.sh:13]: oh. this is not for only building lambda layer? it is to build everything for testing lambda?

File: lambda-layer/src/otel-instrument [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    export OTEL_RESOURCE_ATTRIBUTES=$LAMBDA_RESOURCE_ATTRIBUTES;
+else
+    export OTEL_RESOURCE_ATTRIBUTES="$LAMBDA_RESOURCE_ATTRIBUTES,$OTEL_RESOURCE_ATTRIBUTES";
+fi
+
+# - Enable botocore instrumentation by default
+
+if [ -z ${OTEL_PYTHON_DISABLED_INSTRUMENTATIONS} ]; then
+    export OTEL_PYTHON_DISABLED_INSTRUMENTATIONS="aio-pika,aiohttp-client,aiopg,asgi,asyncpg,boto3sqs,boto,cassandra,celery,confluent-kafka,dbapi,django,elasticsearch,fastapi,falcon,flask,grpc,httpx,jinja2,kafka-python,logging,mysql,mysqlclient,pika,psycopg2,pymemcache,pymongo,pymysql,pyramid,redis,remoulade,requests,sklearn,sqlalchemy,sqlite3,starlette,system-metrics,tornado,tortoiseorm,urllib,urllib3,wsgi"
+fi
```
Comment (mxiamxia) [inline lambda-layer/src/otel-instrument:130]: Could you leave a comment here on the framework we'll only enable for Lambda. `http` and `aws`?

**Review by mxiamxia — APPROVED** [review]

File: lambda-layer/build.sh (hunk 1 above)
Comment (wangzlei) [reply lambda-layer/build.sh:13]: Correct, it build and deploy layer with a sampled app.

**Pull request merged**